### ISSUE TITLE
Getting weird build error MSB4062

### ISSUE THREAD

**Issue by Mitch528** [aspnet/Blazor#1298]
All of a sudden I've been starting to get this build error:

C:\Users\mitch\.nuget\packages\microsoft.aspnetcore.blazor.build\0.5.1\targets\RazorCompilation.targets(184,5): error MSB4062: The "Microsoft.AspNetCore.Razor.Tasks.RazorGenerate" task could not be loaded from the assembly C:\Users\mitch\.nuget\packages\microsoft.aspnetcore.razor.design\2.1.0\build\netstandard2.0\..\..\\tasks\netstandard2.0\Microsoft.AspNetCore.Razor.Tasks.dll. Assembly with same name is already loaded Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask.

The project was running before just fine too. It seems to have started happening once I ran another ASP.NET Core (non-blazor) project. I've tried cleaning/rebuilding the project and clearing nuget locals, but I can't seem to fix it.

Seems to be related to #1107

**Comment by rynowak** [comment]
This looks like a conflict related to assembly versions - can you share the project file of the project that's having an issue?

**Comment by Mitch528** [comment]
@rynowak 

Yeah.

```xml
<Project Sdk="Microsoft.NET.Sdk.Web">

  <PropertyGroup>
    <TargetFramework>netcoreapp2.1</TargetFramework>
    <LangVersion>7.3</LangVersion>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.AspNetCore.App" />
    <PackageReference Include="Microsoft.AspNetCore.Blazor.Server" Version="0.5.1" />
  </ItemGroup>

  <ItemGroup>
    <ProjectReference Include="..\Project.Client\Project.Client.csproj" />
    <ProjectReference Include="..\Project.Shared\Project.Shared.csproj" />
  </ItemGroup>

</Project>
```

```xml
<Project Sdk="Microsoft.NET.Sdk.Web">

  <PropertyGroup>
    <TargetFramework>netstandard2.0</TargetFramework>
    <OutputType>Exe</OutputType>
    <LangVersion>7.3</LangVersion>
    <TypeScriptToolsVersion>2.8</TypeScriptToolsVersion>
  </PropertyGroup>

  <ItemGroup>
    <Content Remove="wwwroot\ts\app.ts" />
    <Content Remove="wwwroot\ts\authentication.ts" />
    <Content Remove="wwwroot\ts\datetimepicker.ts" />
    <Content Remove="wwwroot\ts\dialogs.ts" />
    <Content Remove="wwwroot\ts\dotnet.ts" />
    <Content Remove="wwwroot\ts\dragula.ts" />
    <Content Remove="wwwroot\ts\editor.ts" />
    <Content Remove="wwwroot\ts\interop.ts" />
    <Content Remove="wwwroot\ts\markdown.ts" />
    <Content Remove="wwwroot\ts\tables.ts" />
    <Content Remove="wwwroot\ts\toastr.ts" />
    <Content Remove="wwwroot\ts\trees.ts" />
    <Content Remove="wwwroot\ts\utils.ts" />
  </ItemGroup>

  <ItemGroup>
    <PackageReference Include="Blazor.Extensions.SignalR" Version="0.1.5" />
    <PackageReference Include="BlazorStrap" Version="0.5.0" />
    <PackageReference Include="Microsoft.AspNetCore.Blazor.Browser" Version="0.5.1" />
    <PackageReference Include="Microsoft.AspNetCore.Blazor.Build" Version="0.5.1" />
  </ItemGroup>

  <ItemGroup>
    <ProjectReference Include="..\Project.Shared\Project.Shared.csproj" />
  </ItemGroup>

  <ItemGroup>
    <Folder Include="wwwroot\dist\" />
  </ItemGroup>

  <ItemGroup>
    <TypeScriptCompile Include="wwwroot\ts\app.ts" />
    <TypeScriptCompile Include="wwwroot\ts\authentication.ts" />
    <TypeScriptCompile Include="wwwroot\ts\datetimepicker.ts" />
    <TypeScriptCompile Include="wwwroot\ts\dialogs.ts" />
    <TypeScriptCompile Include="wwwroot\ts\dotnet.ts" />
    <TypeScriptCompile Include="wwwroot\ts\dragula.ts" />
    <TypeScriptCompile Include="wwwroot\ts\editor.ts" />
    <TypeScriptCompile Include="wwwroot\ts\interop.ts" />
    <TypeScriptCompile Include="wwwroot\ts\markdown.ts" />
    <TypeScriptCompile Include="wwwroot\ts\tables.ts" />
    <TypeScriptCompile Include="wwwroot\ts\toastr.ts" />
    <TypeScriptCompile Include="wwwroot\ts\trees.ts" />
    <TypeScriptCompile Include="wwwroot\ts\utils.ts" />
  </ItemGroup>

</Project>
```

I'm actually not so sure if this is a Blazor-specific issue anymore though. I just recently started getting this error trying to build a non-blazor project.

C:\Program Files\dotnet\sdk\NuGetFallbackFolder\microsoft.aspnetcore.razor.design\2.1.1\build\netstandard2.0\Microsoft.AspNetCore.Razor.Design.CodeGeneration.targets(69,5): error MSB4062: The "Microsoft.AspNetCore.Razor.Tasks.RazorTagHelper" task could not be loaded from the assembly C:\Program Files\dotnet\sdk\NuGetFallbackFolder\microsoft.aspnetcore.razor.design\2.1.1\build\netstandard2.0\..\..\\tasks\netstandard2.0\Microsoft.AspNetCore.Razor.Tasks.dll. Assembly with same name is already loaded Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask.

This issue only seems to occur when running it from the command line; it builds fine on Visual Studio. Also, this error is now occuring sporadically, sometimes it will build and sometimes it won't. The problem only starts to appear when I switch from building/running a regular ASP.NET Core to a Blazor project.

**Comment by rynowak** [comment]
Could you try one thing for me? Add a package reference `<PackageReference Include="Microsoft.AspNetCore.Razor.Design" Version="2.1.1" />` to your client project

**Comment by Mitch528** [comment]
Yeah, that seems to have fixed it.

**Comment by Jan-Hettich** [comment]
As of 9/8/18, Razor.Design is at 2.1.2, while AspNetCore.App is at 2.1.3, and I was having this issue until, based on your suggestion, I put explicit references to both package versions:
`    <PackageReference Include="Microsoft.AspNetCore.Razor.Design" Version="2.1.2" />
    <PackageReference Include="Microsoft.AspNetCore.App" Version="2.1.3" />`
Not sure if this is the best way going forward, but it seems to work for now.

**Comment by freerider7777** [comment]
Same thing, I had to add PackageReference 2.1.2 to all projects for dotnet build from command line (VS 2017 builds without errors which is strange)

**Comment by zfilipov** [comment]
I added the package reference to all CSPROJ files for Microsoft.AspNetCore.Razor.Design 2.1.2 but I still get the same error. I am trying to do this through a TFS Build Definition. Any other ideas for this error? As @Mitch528 said, building this with MSBuild works fine, but the dotnet build command throws this error. We are writing a web application using Razor Pages and Razor Libraries. 

**C:\WINDOWS\system32\config\systemprofile\.nuget\packages\microsoft.aspnetcore.razor.design\2.1.2\build\netstandard2.0\Microsoft.AspNetCore.Razor.Design.CodeGeneration.targets(121,5): error MSB4062: The "Microsoft.AspNetCore.Razor.Tasks.RazorGenerate" task could not be loaded from the assembly C:\WINDOWS\system32\config\systemprofile\.nuget\packages\microsoft.aspnetcore.razor.design\2.1.2\build\netstandard2.0\..\..\\tasks\netstandard2.0\Microsoft.AspNetCore.Razor.Tasks.dll. Assembly with same name is already loaded Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask. [csproj file name]**

**Comment by rynowak** [comment]
This will be fixed with the new templates in 0.6.

In general you will run into problems when your solution uses multiple versions of `Microsoft.AspNetCore.Razor.Design` - I'd recommend that every project using `.cshtml` files use `2.1.2`.

**Comment by avjabalpur** [comment]
I got the same error

added this line <PackageReference Include="Microsoft.AspNetCore.Razor.Design" Version="2.1.2" PrivateAssets="All" /> in .csproj

and it seems that the issue got fixed.
